core/cu29_runtime/src/monitoring.rs

```diff
@@ -334,11 +334,12 @@ impl CuDurationStatistics {
     #[inline]
     pub fn record(&mut self, value: CuDuration) {
         let CuDuration(nanos) = value;
-        self.bare.stats.record(nanos).unwrap();
         if self.bare.is_empty() {
+            self.bare.stats.record(nanos).unwrap();
             self.last_value = value;
             return;
         }
+        self.bare.stats.record(nanos).unwrap();
```

Comment:
Can you add a comment to explain the reason behind this commit? I believe it is not very trivial for future users.

Reply:
Ha yeah it tripped me I introduced a regression because once you record it is not empty anymore :)

Reply:
done